ui/cryptomaterial/progressbar.go

```diff
@@ -235,6 +235,20 @@ func (mp *MultiLayerProgressBar) Layout(gtx C, labelWdg layout.Widget) D {
 	)
 }
 
+func (mp *MultiLayerProgressBar) Layout2(gtx C, labelWdg layout.Widget) D {
+	return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			if mp.ShowLedger {
+				return labelWdg(gtx)
+			}
+			return D{}
+		}),
+		layout.Rigid(func(gtx C) D {
+			return layout.Inset{Top: values.MarginPadding24}.Layout(gtx, mp.progressBarLayout)
+		}),
+	)
+}
```

Comment:
2 things.
1. We don't need a second method to do this, this is quite similar to what the Layout method does.
2. Even if we did, `Layout2` would be a bad name - it doesn't tell us how it differs from the `Layout` method and someone who wants to use this method in the future will have to visit the body of this method, to know what it does before they can decide if to use it or not. But no need to rename, since we can remove it completely.

Modify the original `Layout` method like this.

First, add a `MultiLayerProgressBar.ShowProgressBarFirst` boolean property. I assume that's what is different between this new Layout2 method and the original method. One shows the progressbar first, the other shows it last.

Second, rename the `labelWdg` parameter to `additionalWidget`. 

Second, remove `MultiLayerProgressBar.ShowLedger`, it's always true currently. If a caller wants it to be false, the caller should just pass a `nil additionalWidget`.

Finally, inside the original method, do something like this:
```
if additionalWidget == nil {
	// We're only displaying the progress bar, no need for flex layout to wrap it.
	// TODO: Verify if a top padding is necessary if we're only displaying the progressbar.
	return layout.Inset{Top: values.MarginPadding5}.Layout(gtx, mp.progressBarLayout)
}

progressBarTopPadding, otherWidget := values.MarginPadding24, additionalWidget
if mp.ShowProgressBarFirst {
	// reduce the top padding if we're showing the progress bar before the other widget
	progressBarTopPadding = values.MarginPadding5
	otherWidget = func(gtx C) D {
		return layout.Center.Layout(gtx, additionalWidget)
	}
}

flexWidgets := []layout.FlexChild{
	layout.Rigid(func(gtx C) D {
		return layout.Inset{Top: progressBarTopPadding}.Layout(gtx, mp.progressBarLayout)
	}),
	layout.Rigid(otherWidget),
}

if !mp.ShowLabelBeforeProgressBar {
	// Swap the label and progress bar...
	flexWidgets[0], flexWidgets[1] = flexWidgets[1], flexWidgets[0]
}

return layout.Flex{Axis: layout.Vertical}.Layout(gtx, flexWidgets...)
```

Reply:
thanks